crypto/evp/evp_enc.c

```diff
@@ -150,9 +150,12 @@ int EVP_CipherInit_ex(EVP_CIPHER_CTX *ctx, const EVP_CIPHER *cipher,
  skip_to_init:
 #endif
     /* we assume block size is a power of 2 in *cryptUpdate */
-    OPENSSL_assert(ctx->cipher->block_size == 1
-                   || ctx->cipher->block_size == 8
-                   || ctx->cipher->block_size == 16);
+    if (!ossl_assert(ctx->cipher->block_size == 1
+                     || ctx->cipher->block_size == 8
+                     || ctx->cipher->block_size == 16)) {
+        EVPerr(EVP_F_EVP_CIPHERINIT_EX, ERR_R_INTERNAL_ERROR);
```

Comment:
Same argument as above, assert alone should be sufficient. BTW, the fact that error is pushed kind of emphasizes point behind "unless one start arguing that it's indication of memory corruption". I mean what application is actually supposed to do? You kind of say "yeah, you know, this structure is all screwed up, but here is error, try not to crash." I mean if you are actually convinced that memory is corrupted, you definitely should crash.